### ISSUE TITLE
Local audio not smooth for me....

### ISSUE THREAD

**Issue by frohro** [n1gp/cudaSDR#1]
Hi All,

Having local audio (on the computer) is a great new feature, especially for Hermes-Lite, however for me I find that it is not smooth.  It sounds a little choppy, and not completely clear as a result.

73,

Rob
KL7NA

**Comment by n1gp** [comment]
Hi Rob,

What is the environment that you are running on, PC arch / speed, Linux distro / version, also are you running with pulse audio enabled?

-Rick / N1GP

**Comment by frohro** [comment]
Hi Rick,

I'm using Ubuntu 14.10 on a Dell E6420 (i7 with 8 GB of RAM, NVIDIA 
NVS4200M).  I'm using Pulse Audio.

Does that help?

73,

Rob
KL7NA

On 04/07/2015 01:41 PM, Rick Koch wrote:

> Hi Rob,
> 
> What is the environment that you are running on, PC arch / speed, 
> Linux distro / version, also are you running with pulse audio enabled?
> 
> -Rick / N1GP
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/n1gp/cudaSDR/issues/1#issuecomment-90726100.

## 

Rob Frohne, PhD, PE
Professor
EF Cross School of Engineering
Walla Walla University
100 SW 4th Street
College Place, WA 99324
(509) 527-2075

**Comment by n1gp** [comment]
Yes, that is helpful in that I know you are running a system similar to my own.
The code is currently hard coded to assign the alsa device "default" as the sound
output device for the program. Also the default sound rate is 48Khz.

You can change theses in the "src/DataEngine/soundout.cpp" file. If you look around
line 152 of that file you can change an #if 0 section of code so that it will print out all
your sound output devices to the cuSDR.log file. That may be handy to see what you
can set it to.

I plan to add dialog menus to change these in the future, and maybe also use a pulse audio
based solution.

BTW this audio code was leveraged from Moe Wheatley's CuteSDR.

**Comment by frohro** [comment]
Hi Rick,

Well, I wont and changed the #if 0 to #if 1 and ran things again, and 
they seem to be working for me now.  I also noted a lot less latency 
when I tune the frequency.  I'm not sure what was going on before.

73,

Rob
KL7NA

On 04/08/2015 06:18 AM, Rick Koch wrote:

> Yes, that is helpful in that I know you are running a system similar 
> to my own.
> The code is currently hard coded to assign the alsa device "default" 
> as the sound
> output device for the program. Also the default sound rate is 48Khz.
> 
> You can change theses in the "src/DataEngine/soundout.cpp" file. If 
> you look around
> line 152 of that file you can change an #if 0 section of code so that 
> it will print out all
> your sound output devices to the cuSDR.log file. That may be handy to 
> see what you
> can set it to.
> 
> I plan to add dialog menus to change these in the future, and maybe 
> also use a pulse audio
> based solution.
> 
> BTW this audio code was leveraged from Moe Wheatley's CuteSDR.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/n1gp/cudaSDR/issues/1#issuecomment-90911983.

## 

Rob Frohne, PhD, PE
Professor
EF Cross School of Engineering
Walla Walla University
100 SW 4th Street
College Place, WA 99324
(509) 527-2075
